### PR TITLE
Ameen's Orbital FW Onboarding

### PR DESCRIPTION
**Pull request by AmeenDurani**
# New Changes
- Updated and completed driver functions along with OS and thermal management functions

# Testing
- Verified tests via Unit tests and Integration Tests

### CODE REVIEW

File: lm75bd/lm75bd.h [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #include <stdint.h>
 
 /* LM75BD I2C Device Address */
-#define LM75BD_OBC_I2C_ADDR /* Define the address here */
+#define LM75BD_OBC_I2C_ADDR 0b1001111U/* Define the address here */
```
Comment (Gjjjiang) [inline lm75bd/lm75bd.h]: Hexadecimal is usually used for readability purposes

File: services/thermal_mgr/thermal_mgr.c [hunk 2]
```diff
@@ -43,18 +43,38 @@ void initThermalSystemManager(lm75bd_config_t *config) {
...
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+
+  float temp;
+  thermal_mgr_event_t buff;
+
   while (1) {
-    
+   xQueueReceive(thermalMgrQueueHandle, &buff, 1);
```
Comment (Gjjjiang) [inline services/thermal_mgr/thermal_mgr.c]: We would want the thermal manager to not run until an event is received. Use portMAX_DELAY  to indefinitely block the task until it receives an event.

File: lm75bd/lm75bd.c [hunk 3]
```diff
@@ -27,7 +27,25 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
...
+  i2cSendTo(LM75BD_OBC_I2C_ADDR, pointerReg, 1);
+
+  uint8_t twosCOMP[2] = {0};
+  int16_t decimalREP = 0;
+  i2cReceiveFrom(LM75BD_OBC_I2C_ADDR, twosCOMP, 2);
+  decimalREP = twosCOMP[0];
+  decimalREP=(decimalREP << 8) | twosCOMP[1];
+
+  if((twosCOMP[0] & 0b10000000) == 0b10000000){ //two's complement
+    decimalREP = ((~decimalREP)>>5)+0b0000000000000001;
```
Comment (Gjjjiang) [inline lm75bd/lm75bd.c]: +1 is fine, no need for its binary representation.

File: lm75bd/lm75bd.c [hunk 4]
```diff
@@ -27,7 +27,25 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
-  
+
+  uint8_t pointerReg[1] = {0};
```
Comment (Gjjjiang) [inline lm75bd/lm75bd.c]: 1 element large array is unnecessary. Just use uint8_t set to the pointer value.

File: lm75bd/lm75bd.c [hunk 5]
```diff
@@ -27,7 +27,25 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
...
+  i2cSendTo(LM75BD_OBC_I2C_ADDR, pointerReg, 1);
```
Comment (Gjjjiang) [inline lm75bd/lm75bd.c]: i2cSendTo(LM75BD_OBC_I2C_ADDR, &pointerReg, 1);

Comment (Gjjjiang) [inline lm75bd/lm75bd.c]: Also, make sure to check the return value of the functions. i2cSendTo returns an error code. You can use RETURN_IF_ERROR_CODE macro found in sys/logging/logging.h.

File: services/thermal_mgr/thermal_mgr.c (hunk 2 above)
Comment (Gjjjiang) [inline services/thermal_mgr/thermal_mgr.c]: Make sure to check the return value of xQueueReceive

File: services/thermal_mgr/thermal_mgr.h [hunk 6]
```diff
@@ -5,6 +5,7 @@
 
 typedef enum {
   THERMAL_MGR_EVENT_MEASURE_TEMP_CMD,
+  InteruptSqn,
```
Comment (Gjjjiang) [inline services/thermal_mgr/thermal_mgr.h]: Make sure to be consistent with the naming scheme used.

**Review by Gjjjiang — CHANGES_REQUESTED** [review]
Overall good work! There's just some small changes needed.

File: lm75bd/lm75bd.h (hunk 1 above)
Comment (AmeenDurani) [reply lm75bd/lm75bd.h]: changing to -> #define LM75BD_OBC_I2C_ADDR 0x4FU

File: services/thermal_mgr/thermal_mgr.c (hunk 2 above)
Comment (AmeenDurani) [reply services/thermal_mgr/thermal_mgr.c]: changing to xQueueReceive(thermalMgrQueueHandle, &buff, portMAXDELAY);

File: lm75bd/lm75bd.c [hunk 7]
```diff
@@ -27,7 +27,26 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
-  
+  error_code_t errCode;
+  uint8_t pointerReg = 0;
```
Comment (Gjjjiang) [inline lm75bd/lm75bd.c]: Magic number

File: lm75bd/lm75bd.c [hunk 8]
```diff
@@ -27,7 +27,26 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
...
+  error_code_t _ret = i2cSendTo(LM75BD_OBC_I2C_ADDR, &pointerReg, 1);
+  RETURN_IF_ERROR_CODE(_ret);
+  uint8_t twosCOMP[2] = {0};
```
Comment (Gjjjiang) [inline lm75bd/lm75bd.c]: Make sure to make a define constant for the array size.

File: lm75bd/lm75bd.c [hunk 9]
```diff
@@ -27,7 +27,26 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
...
+  int16_t decimalREP = 0;
+  _ret=i2cReceiveFrom(LM75BD_OBC_I2C_ADDR, twosCOMP, 2);
```
Comment (Gjjjiang) [inline lm75bd/lm75bd.c]: replace 2 with the define constant array size

File: lm75bd/lm75bd.c [hunk 10]
```diff
@@ -27,7 +27,26 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
...
+  RETURN_IF_ERROR_CODE(_ret);
+  decimalREP = twosCOMP[0];
+  decimalREP=(decimalREP << 8) | twosCOMP[1];
+
+  if((twosCOMP[0] & 0b10000000) == 0b10000000){ //two's complement
```
Comment (Gjjjiang) [inline lm75bd/lm75bd.c]: Magic number. Make to make a define constant for the bit mask. (maybe call it TEMP_SIGN_MASK or something similar)

**Review by Gjjjiang — CHANGES_REQUESTED** [review]
3 more small changes.

File: lm75bd/lm75bd.c [hunk 11]
```diff
@@ -27,7 +27,31 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
-  
+
+  error_code_t errCode;
+  #define POINTER_VALUE 0 //pointer bit value B1 and B0
```
Comment (Gjjjiang) [inline lm75bd/lm75bd.c]: move this to line 12 and call it LM75BD_REG_TEMP or something similar. Define constants usually appear near the start of the file and not in functions

File: lm75bd/lm75bd.c [hunk 12]
```diff
@@ -27,7 +27,31 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
...
+
+  #define ARRAY_SIZE 2 //array size constant
+  uint8_t twosCOMP[ARRAY_SIZE] = {0};
+  int16_t decimalREP = 0;
+  _ret=i2cReceiveFrom(LM75BD_OBC_I2C_ADDR, twosCOMP, ARRAY_SIZE);
+  RETURN_IF_ERROR_CODE(_ret);
+
+  decimalREP = twosCOMP[0];
+  decimalREP=(decimalREP << 8) | twosCOMP[1];
+  #define TEMP_SIGN_MASK 0b10000000
```
Comment (Gjjjiang) [inline lm75bd/lm75bd.c]: same thing as above

**Review by Gjjjiang — CHANGES_REQUESTED** [review]
2 small changes, otherwise lgtm

**Review by Gjjjiang — APPROVED** [review]
lgtm

**Pull request closed without merging**